basic/package.json

```diff
@@ -0,0 +1,4 @@
+{
+    "main": "../dist/flavors/basic.js",
```

Comment:
The `exports` mechanism in the `top-level` package.json ends up creating a source tree that is disjoint from the usual `dist/` structure. The standard guidance for defining sub-package exports only works if each of this directory has no dependencies on the rest of the source, which is not the case here -- we want to define a vanity export that exists *within* the source.

I eventually found a reference in the [thread](https://github.com/microsoft/TypeScript/issues/33079) that suggested defining a minimal `package.json` file to help with this mapping.